src/Features/CSharp/Portable/CodeFixes/AddParameterXmlDocumentation/AddParameterXmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,316 @@
+                    diagnostic);
+            }
+            else if (diagnostic.Id == CS1712)
+            {
+                var parameterList = FindFirstAncestorTypeParameterItem(root.FindToken(diagnosticSpan.Start).Parent);
+                if (parameterList == null) { return; }
+
+                var parameterSyntax = root.FindToken(diagnosticSpan.Start).GetAncestor<TypeParameterSyntax>();
+                if (parameterSyntax == null) { return; }
+                var parameterName = parameterSyntax.Identifier.ValueText;
+
+                XmlNameAttributeSyntax adjacentAttribute;
+                bool adjacentAttributeIsBefore;
+                FindAdjacentParameterDocumentation(
+                    parameterName,
+                    allParameterNames: parameterList.Parameters.Select(p => p.Identifier.ValueText),
+                    tokenWithDocumentationComment: parameterList.Parent.GetFirstToken(),
+                    elementKind: XmlNameAttributeElementKind.TypeParameter,
+                    adjacentNameAttribute: out adjacentAttribute,
+                    adjacentAttributeIsBefore: out adjacentAttributeIsBefore);
+
+                if (adjacentAttribute == null) { return; }
+
+                context.RegisterCodeFix(
+                    new AddParameterXmlDocumentationCodeAction(context.Document, parameterName, adjacentAttribute,
+                        adjacentAttributeIsBefore, isTypeParameter: true),
+                    diagnostic);
+            }
+        }
+
+        /// <summary>
+        /// Finds a xml documentation tag with a name attribute at which another a new tag can be inserted.
+        /// </summary>
+        /// <param name="parameterName">The parameter of the new tag</param>
+        /// <param name="allParameterNames">All the parameter of which <paramref name="parameterName"/> is one.</param>
+        /// <param name="tokenWithDocumentationComment"></param>
+        /// <param name="elementKind">Specifies which elements can be found.</param>
+        /// <param name="adjacentNameAttribute">The name attribute of the adjacent parameter. If null no parameter was found.</param>
+        /// <param name="adjacentAttributeIsBefore">Indicates if the found adjacent attribute is before or after the parameter.</param>
+        private static void FindAdjacentParameterDocumentation(string parameterName, IEnumerable<string> allParameterNames, SyntaxToken tokenWithDocumentationComment,
+            XmlNameAttributeElementKind elementKind, out XmlNameAttributeSyntax adjacentNameAttribute, out bool adjacentAttributeIsBefore)
+        {
+            FindAdjacentParameterDocumentationWalker.FindAdjacentParameterDocumenteden(parameterName, allParameterNames, tokenWithDocumentationComment,
+                elementKind, out adjacentNameAttribute, out adjacentAttributeIsBefore);
+        }
+
+        private class FindAdjacentParameterDocumentationWalker : CSharpSyntaxWalker
```

Comment:
On second thought -- is a `SyntaxWalker` overkill here? There's a lot of code here to prepare to use this, but it seems like walking through an XML doc comment's param nodes isn't all that complicated. After all, they all have to be at the top-level (i.e. they're never nested). Can't you just loop over the elements in the doc comment?